src/format/KdbxXmlReader.cpp

```diff
@@ -814,26 +814,14 @@ void KdbxXmlReader::parseEntryString(Entry* entry)
             QXmlStreamAttributes attr = m_xml.attributes();
             value = readString();
 
+            if (m_error) {
+                raiseError(tr("Unable to decrypt entry string"));
+                continue;
+            }
+
             bool isProtected = attr.value("Protected") == "True";
             bool protectInMemory = attr.value("ProtectInMemory") == "True";
```

Comment:
You should remove this here. Both Protected and ProtectInMemory should be handled by the readString() method. That method should have additional output parameters to indicate whether the attribute was protected and should be protected in memory.

Reply:
Not sure how to handle that. Two output references? But readString and readBinary are called without requiring the result. Should I add 0 and 2 parameter overloads?

Reply:
Two output references, yes. You may add overloads, but I ideally, all callers of this method should use those parameters.

Reply:
Dozens of occurences and (iirc) only 2 need the output, so I chose the overloaded version.

Reply:
OK.